Adding a quick tip in debug hover mode

sseelam2 opened the pull request:
This PR fixes #107697
As @isidorn suggested, I added a common HTML element `tip` which is used by both `complex-value` and `value` container.
@misolori Let me know if the design and the text -`'Quick tip: Hold alt key to switch to editor hover'` can be changed.

isidorn commented on src/vs/workbench/contrib/debug/browser/debugHover.ts:
```diff
@@ -116,10 +118,10 @@ export class DebugHoverWidget implements IContentWidget {
 		this.valueContainer = $('.value');
 		this.valueContainer.tabIndex = 0;
 		this.valueContainer.setAttribute('role', 'tooltip');
+		this.tip.innerText = 'Quick tip: Hold alt key to switch to editor hover';
```
I would prefer the following text:
'Hold alt key to switch to editor language hover'

isidorn replied:
And make sure to call `nls.localize` on the string so we get it translated in different lagnuages

isidorn commented on src/vs/workbench/contrib/debug/browser/media/debugHover.css:
```diff
@@ -32,6 +32,17 @@
+	word-break: normal;
+	height: 18px;
+	white-space: pre;
+	border-top: 1px solid rgba(128, 128, 128, 0.35);
```
We usualy try not to introduce colors directly into `css` so that they can be customised.

isidorn commented:
Good work. 
I have checked out the code and have commented directly in the code.
Can you please attach a picture of how the hover looks with the tip. And once we have that we can decide if we want to go futher with this.

Please note that next 2 weeks I am on vacation, so I might be slower when responding.

misolori commented:
Design looks good 👍 

![image](https://user-images.githubusercontent.com/35271042/95118488-3f199380-06ff-11eb-8a70-3ff05c84417d.png)

@isidorn the css is copied from this line so we should just make this themeable in a future change:
https://github.com/microsoft/vscode/blob/7a01f2077dee045d51760a60bad78213d1f88ac3/src/vs/workbench/contrib/debug/browser/media/debugHover.css#L32

misolori commented on src/vs/workbench/contrib/debug/browser/media/debugHover.css:
```diff
@@ -32,6 +32,17 @@
 	border-bottom: 1px solid rgba(128, 128, 128, 0.35);
 }
 
+.tip {
```
I would just add the `.tip` class to the same rules from the `.title` class here so we're not duplicating values

https://github.com/microsoft/vscode/blob/7a01f2077dee045d51760a60bad78213d1f88ac3/src/vs/workbench/contrib/debug/browser/media/debugHover.css#L22

isidorn commented:
@sseelam2 do you plan to address our comments or should I go ahead and close this PR?
Thanks!

justingrant commented:
@sseelam2 - This looks like a great improvement!  Thanks for doing this. Are you planning to address the requested changes?  It'd be good to see this PR get finished.

sseelam2 commented:
> @sseelam2 - This looks like a great improvement! Thanks for doing this. Are you planning to address the requested changes? It'd be good to see this PR get finished.

Yes! I am so sorry. I missed the emails. I will work on addressing the comments today.

isidorn commented:
Great!

sseelam2 commented:
I have addressed the comment. I took the below screenshot after the recent changes. Please let me know if anything else needs be done! :)

![image](https://user-images.githubusercontent.com/42684680/98600851-0f782100-2293-11eb-9a7d-75b8e6cb698a.png)

isidorn commented:
Thanks for continuing to work on this.
There are still some issues, and I have commented directly in the code. Thanks

isidorn commented on src/vs/workbench/contrib/debug/browser/media/debugHover.css:
```diff
@@ -19,7 +19,7 @@
 }
 
-.monaco-editor .debug-hover-widget .complex-value .title {
+.monaco-editor .debug-hover-widget .complex-value .title .tip {
```
This is not good. Since you have made the previous rule more specific, thus it applies to elements that have a class `tip` as a child of `title`. However there are no such elements.
What I think you want is this rule to be aplied to both .title and .tip for that you have to do something like this

```
.monaco-editor .debug-hover-widget .complex-value .title, 
.monaco-editor .debug-hover-widget .complex-value .tip {
```

isidorn commented on src/vs/workbench/contrib/debug/browser/debugHover.ts:
```diff
@@ -116,10 +118,10 @@ export class DebugHoverWidget implements IContentWidget {
 		this.valueContainer = $('.value');
 		this.valueContainer.tabIndex = 0;
 		this.valueContainer.setAttribute('role', 'tooltip');
+		this.tip.innerText = nls.localize('quickTip', 'Quick tip: Hold alt key to switch to editor language hover');
```
Please remove `"Quick tip:"` part from the message

isidorn commented:
I went ahead and made the changes. Thank you for the PR which I used as inspiration. 
Since this is now tackled I am closing this
